### PR TITLE
Make the `fieldValuesWidget` a functional component

### LINKED ISSUE #32158: fieldValuesWidget treats model fields differently than underlying table
### Describe the bug

Models and Underlying tables have their field values treated differently in the filter widget in the notebook. probably has something to do with virtual fields. There's also some potential weirdness around how fetching field values mutates the field entity cache in redux.

- models don't have checkboxes
- re-selecting a model field the second time shows no choices

products table: 
table-products-fields

model based on the products table:
model-field-search

### To Reproduce

filter any model with fields that have listable field values in the notebook (see gifs above)

### Expected behavior

the same field should appear the same regarless of where it's in a model or a table

### Logs

_No response_

### Information about your Metabase installation

```JSON
v.47.x (probably been around for a bit)
```

### Severity

very not severe

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iethree**
### Description

The Field Values Widget is big, ugly, and huge source of bugs. It will also need some updates in order to be used by MetabaseLibV2.

Along the way found https://github.com/metabase/metabase/issues/32158 and fixes https://github.com/metabase/metabase/issues/32158

This starts that process by converting it to a functional component and doing a little bit of organizing its helper functions and tests. The only real improvement here is that the component props are enumerated.

**Next**: convert to typescript
**After that**: make it better

### How to verify

You can test this component in the app in various places:
- filter dropdowns from table headings in question mode
- search values in the filter modal
- dashboard parameter/filter input

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx [hunk 1]
```diff
@@ -68,405 +74,364 @@ function mapStateToProps(state, { fields = [] }) {
...
-      const { fetchFieldValues } = this.props;
-      await Promise.all(
-        getNonVirtualFields(this.props.fields).map(field =>
-          fetchFieldValues(field.id),
-        ),
+      const nonVirtualFields = getNonVirtualFields(fields);
+
+      const results = await Promise.all(
+        nonVirtualFields.map(field => fetchFieldValuesProp({ id: field.id })),
       );
```
Comment (iethree) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx:187]: I had to tweak this a bit. The previous logic relied on mutations to `this.props.fields` in the middle of this function. Instead, this relies entirely on the return value from the entity loader.

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx [hunk 2]
```diff
@@ -53,7 +59,7 @@ const optionsMessagePropTypes = {
 
 const mapDispatchToProps = {
```
Comment (EmmadUsmani) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx:60]: I think we can get rid of the `mapDispatchToProps` with `useDispatch` + directly importing the actions. Same for `mapStateToProps`, we can just use `useSelector` and then we can drop the `connect` HoC.

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js [hunk 3]
```diff
@@ -287,132 +218,6 @@ describe("FieldValuesWidget", () => {
     });
   });
 
-  describe("searchField", () => {
```
Comment (EmmadUsmani) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js:290]: Why was this entire suite removed?

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx (hunk 2 above)
Comment (iethree) [reply frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx:60]: absolutely, that's on my toDo list, but I wanted to keep the PRs as small as possible and do the minimum needed to make this a functional component.

**Review by EmmadUsmani — APPROVED** [review]
Looks good 👍

**Pull request merged**